### PR TITLE
Wireapp/4.6.2 federation with drain

### PR DESCRIPTION
**Pull request by sgodin**
Add new Drain feature to coturn

-invoke Drain mode by signaling SIGUSR1 or sending "drain" command via coturn cli (telnet)
-when Drain mode is enabled, existing allocations are allowed to continue operation without issue
-any new allocation attempts will get a 300 response 
  -300 (Try Alternate): The client should contact an alternate server for this request.
-when last allocation goes away the coturn server will complete it's shutdown and the process will terminate

### CODE REVIEW

File: src/apps/relay/mainrelay.c [hunk 1]
```diff
@@ -3963,4 +3967,11 @@ static void shutdown_handler(evutil_socket_t sock, short events, void *args) {
   UNUSED_ARG(args);
 }
 
+static void drain_handler(evutil_socket_t sock, short events, void *args) {
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "Draining then terminating on signal %d\n", sock);
+  turn_params.drain_turn_server = 2;
```
Comment (z-dule) [inline src/apps/relay/mainrelay.c:3972]: Maybe use a constant instead of the hard-coded "2"?

File: src/apps/relay/netengine.c [hunk 2]
```diff
@@ -1613,6 +1613,33 @@ void run_listener_server(struct listener_server *ls) {
       }
     }
 
+    // Check if we are draining
+    if(turn_params.drain_turn_server == 2) {
```
Comment (z-dule) [inline src/apps/relay/netengine.c:1617]: Maybe use a constant instead of the hard-coded "2"?

File: src/apps/relay/turn_admin_server.c [hunk 3]
```diff
@@ -924,6 +927,12 @@ static int run_cli_input(struct cli_session *cs, const char *buf0, unsigned int
         myprintf(cs, "%s\n", str);
         close_cli_session(cs);
         ret = -1;
+      } else if (strcmp(cmd, "drain") == 0) {
+        addr_debug_print(1, &(cs->addr), "Drain command received from CLI user");
+        const char *str = "TURN server is draining then shutting down";
+        myprintf(cs, "%s\n", str);
+        close_cli_session(cs);
+        turn_params.drain_turn_server = 2;
```
Comment (z-dule) [inline src/apps/relay/turn_admin_server.c:935]: Maybe use a constant instead of the hard-coded "2"?

Comment (sgodin) [reply src/apps/relay/turn_admin_server.c]: I used an enum instead, no more magic 2 number.  :)

**Pull request merged**